### PR TITLE
[aoc-collector non-root user]: Create and use a new user for the container image

### PR DESCRIPTION
**Pull request by PaurushGarg**
**Description:** <Describe what has changed.>

This change is cherry-picked from dev branch commit: `bc5fb1c8`

Adding `aoc` user to docker file for dev branch as done in [this](https://github.com/aws-observability/aws-otel-collector/pull/1549) PR.
Haven't removed [RUN_IN_CONTAINER](https://github.com/aws-observability/aws-otel-collector/blob/main/pkg/extraconfig/container.go#L21) environment variable as it is getting used at [this](https://github.com/aws-observability/aws-otel-collector/blob/8343b25181940ee656138f42986462500f623672/pkg/logger/logger.go#L41) place as well.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/aws-observability/aws-otel-collector/issues/2222

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rapphil — CHANGES_REQUESTED** [review]
Blocking this change until we are ready to merge it.

**Review by bryan-aguilar — APPROVED** [review]

**Pull request merged**
